static/app/components/checkbox.tsx

```diff
@@ -1,16 +1,37 @@
 import {useEffect, useRef} from 'react';
+import styled from '@emotion/styled';
+import {useHover, usePress} from '@react-aria/interactions';
+import {mergeProps} from '@react-aria/utils';
+
+import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import {IconCheckmark, IconSubtract} from 'sentry/icons';
+import commonTheme, {FormSize} from 'sentry/utils/theme';
 
 type CheckboxProps = React.InputHTMLAttributes<HTMLInputElement>;
 
-interface Props extends Omit<CheckboxProps, 'checked'> {
+interface Props extends Omit<CheckboxProps, 'checked' | 'size'> {
   /**
    * Is the checkbox active? Supports 'indeterminate'
    */
   checked?: CheckboxProps['checked'] | 'indeterminate';
+  /**
+   *
+   */
+  size?: FormSize;
 }
 
-const Checkbox = ({checked = false, ...props}: Props) => {
+type CheckboxConfig = {borderRadius: string; box: string; icon: string};
+
+const checkboxSizeMap: Record<FormSize, CheckboxConfig> = {
+  xs: {box: '12px', icon: '10px', borderRadius: '4px'},
+  sm: {box: '16px', icon: '12px', borderRadius: '4px'},
+  md: {box: '22px', icon: '16px', borderRadius: commonTheme.borderRadius},
+};
+
+const Checkbox = ({checked = false, size = 'sm', ...props}: Props) => {
   const checkboxRef = useRef<HTMLInputElement>(null);
+  const {isPressed, pressProps} = usePress({isDisabled: props.disabled});
+  const {isHovered, hoverProps} = useHover({isDisabled: props.disabled});
```

Comment:
Do we have to have this state, or is there a way to do this with pure css?

Reply:
I guess InteractionStateLayer needs it huh

Reply:
cc @vuluongj20 on this.

Maybe it's fine, my gut just generally tells me don't use javascript when you don't need it. But also we already use a lot of JS so 🤷

Reply:
`InteractionStateLayer` has a fallback: if the `isHovered`/`isPressed` prop isn't defined, then it will use the (pure css) `:hover`/`:active` state of its immediate parent.

Here that fallback doesn't work because `InteractionStateLayer`'s parent — `StyledCheckbox` — has `pointer-events: none`. @malwilley Why did you add a `pointer-events: none` to it?

Reply:
@vuluongj20 the actual interact-able element here is the `input` which has `opacity: 0` and is behind the styled checkbox div (which is only there for presentation purposes). Click events will go to the `input`.  Since the styles need to added to the presentational div and not the `input` the default styles don't work in this use case.

Reply:
@malwilley Hmm how about we move `InteractionStateLayer` up a level, making it `Wrap`'s child (and also add a border radius to `Wrap`)?

Reply:
OR we can just keep using `useHover` lol. There are [benefits to using it](https://react-spectrum.adobe.com/react-aria/useHover.html#features) over pure css.

Reply:
Good thinking, I was able to move it out! CSS only now https://github.com/getsentry/sentry/pull/42583/commits/3f6871800d75f9495523b701218d0d509286b53b